rust/src/graph/higher_order_queries.rs

```diff
@@ -327,4 +327,57 @@ mod tests {
             ])
         );
     }
+
+    #[test]
+    fn test_generate_module_permutations_two_closed_layers() {
```

Comment:
I think we should have Python tests for this too (or instead of) - one of the principles of the project is that the Python tests are pretty comprehensive.

Reply:
Okay sure - python tests extended now.